cmd/iam.go

```diff
@@ -1388,34 +1426,25 @@ func (sys *IAMSys) IsAllowedSTS(args iampolicy.Args, parentUser string) bool {
 		}
 		policies = newMappedPolicy(sys.rolesMap[arn]).toSlice()
 	} else {
-		// If roleArn is not used, we fall back to using policy claim
-		// from JWT.
-		policySet, ok := args.GetPolicies(iamPolicyClaimNameOpenID())
-		if !ok {
-			// When claims are set, it should have a policy claim field.
-			return false
-		}
-		// When claims are set, it should have policies as claim.
-		if policySet.IsEmpty() {
-			// No policy, no access!
-			return false
-		}
-
-		// If policy is available for given user, check the policy.
-		mp, ok := sys.store.GetMappedPolicy(args.AccountName, false)
+		// Lookup the parent user's mapping if there's no role-ARN.
+		mp, ok := sys.store.GetMappedPolicy(parentUser, false)
 		if !ok {
-			// No policy set for the user that we can find, no access!
-			return false
-		}
+			// TODO: Only need to handle behavior for STS
```

Comment:
Can you add 'Deprecated in Dec 2021' ?

It will be easier to clean the code later

Reply:
Done.